service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -252,6 +252,31 @@ public JsonElement getVersions() {
         return parser.parse(versions.toString());
     }
 
+    /**
+     * Used to display the available message protocol list and their edition names.
+     *
+     * @return json with service names and respective edition details.
+     */
+    @ApiOperation(value = "To get the available message protocol list and edition names", response = String.class)
+    @RequestMapping(value = "/message_protocols", method = RequestMethod.GET)
+    public JsonElement getMessageProtocols() {
+        JsonArray array = new JsonArray();
+        JsonObject semanticsObject = new JsonObject();
+        JsonObject eiffel3Object = new JsonObject();
+        for(MsgService service : msgServices) {
+            if(service.getServiceName().equals("eiffelsemantics")) {
```

Comment:
Why you need two different JsonObjects? No need to have if else statements here. You can add jsonObjects to array directly here.